rfd/0166-tpm-joining.md

```diff
@@ -0,0 +1,390 @@
+---
+authors: @strideynet (noah@goteleport.com)
+state: draft
+---
+
+# RFD 0166 - TPM Joining
+
+## Required Approvers
+ 
+* Engineering: (@zmb33 || @codingllama)
+* Security: (@reedloden || @jentfoo)
+* Product: (@xinding33 || @klizhentas)
+
+# What
+
+Introduce a new join method that allows a Bot or Agent running on a host with a
+TPM to securely join a Teleport cluster without using a shared secret.
```

Comment:
> TPM

This could work for Secure Enclave devices too, although I would guess TPM-bots are likely much more common.

Reply:
Yeah - we could certainly apply a similar principal, but I think it would have to be somewhat distinct given the difference in the ceremonies available.